rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+                $ref: '#/components/schemas/NoSuchTableError'
+              example:
+                '{ error: { message: "Table does not exist", type: "NoSuchTableException", code: 41202 }'
+        '409':
+          description: The new table identifier, the to table rename to, already exists.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableAlreadyExistsError'
+              example:
+                '{ error: { message: "Namespace already exists", type: "AlreadyExistsException", code: 40902 }'
+  /v1/namespaces/{namespace}/properties:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored metadata properties for a given namespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application:
+              schema:
+                $ref: '#/components/schemas/GetNamespaceResponse'
+        '417':
+          description: Namespace not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchNamespaceError'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 41701 }'
+    put:
+      tags:
+        - Catalog API
+      summary: Add or overwrite properties to an existing namespace
+      operationId: setNamespaceProperties
+      description:
+        Adds propertiess for a namespace. This will overwrite any existing properties,
+        and merge with the others.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RemovePropertiesRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            'application/json':
+              schema:
+                type: boolean
+        '417':
+          description: Namespace not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchNamespaceError'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 41701 }'
+        '409':
+          description: Namespace already exists
+          content:
+            applicaton/json:
+              schema:
+                $ref: '#/components/schemas/NamespaceAlreadyExistsError'
+              example:
+                '{ error: { message: "Namespace already exists", type: "AlreadyExistsException", code: 40901 }'
+    post:
+      tags:
+        - Catalog API
+      summary: Overwrite a namespace's properties with a new set of properties
```

Comment:
This is actually not an overwrite operation. The `set` operation sets just the given properties and leaves the rest unchanged.

You might consider copying the documentation from the `SupportsNamespaces` and `Catalog` interfaces. for `setProperties`, we have "Properties that are not in the given map are not modified or removed by this method."

Reply:
Also, I just remembered that we don't want to have an overwrite operation because it seems idempotent. If we expect concurrent writers, then we want it to be able to retry. That means having good semantics that allow retries. Merging properties can be retried, but overwriting could easily drop properties that were added concurrently.